scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,862 @@
+
+    }
+
+    public AOSynchronization(String statusFileDirectoryPath) {
+        compilerConfiguration = new CompilerConfiguration();
+        compilerConfiguration.setTargetBytecode(CompilerConfiguration.JDK8);
+        shell = new GroovyShell(this.getClass().getClassLoader(), compilerConfiguration);
+        this.statusFileDirectory = new File(statusFileDirectoryPath);
+        if (!statusFileDirectory.exists()) {
+            statusFileDirectory.mkdirs();
+        }
+        statusFile = new File(statusFileDirectory, DEFAULT_STORE_SESSION_NAME);
+        inMemoryChannels = new HashMap<>();
+
+    }
+
+    /**
+     * Initialize the Record Manager.
+     *
+     * Load the persisted channel from disk at configured location.
+     *
+     * Delete the persisted database in case a serious error occurred when loading the database.
+     * @param firstAttempt first attempt at loading the persisted channels
+     * @return true if the operation succeeded
+     * @throws IllegalStateException when an error occurred during database loading, even after cleaning it
+     */
+    private boolean init(boolean firstAttempt) {
+        close();
+        try {
+            logger.info("Loading Record Manager from file : " + statusFile);
+            recordManager = RecordManagerFactory.createRecordManager(statusFile.getCanonicalPath());
+            persistedChannels = recordManager.hashMap(STATUS_RECORD_NAME);
+            logger.info("Content of persisted store : " + persistedChannels);
+            // This empty loop triggers InvalidClassException in case of serial version uid problems
+            for (Map.Entry<String, Channel> tempChannels : persistedChannels.entrySet())
+                ;
+            recordManager.commit();
+            if (!firstAttempt) {
+                logger.info("Loading of job database successful after clean.");
+            }
+        } catch (Throwable e) {
+            if (firstAttempt) {
+                logger.error("Error occurred when loading channels database " + statusFile.getAbsolutePath() +
+                             ", now cleaning it and retrying.", e);
+                cleanDataBase();
+                init(false);
+            } else {
+                close();
+                throw new IllegalStateException("Error when loading database (even after cleaning it): " +
+                                                statusFile.getAbsolutePath(), e);
+            }
+        }
+        return true;
+    }
+
+    public boolean isStarted() {
+        return true;
+    }
+
+    public boolean cleanDataBase() {
+        close();
+        logger.info("Cleaning database");
+
+        // delete all db files
+        File[] dbChannelFiles = statusFileDirectory.listFiles((dir,
+                name) -> name.startsWith(DEFAULT_STORE_SESSION_NAME));
+        for (File file : dbChannelFiles) {
+            try {
+                logger.info("Deleting " + file);
+                file.delete();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use "java.nio.Files#delete" here for better messages on error conditions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Do something with the "boolean" value returned by "delete". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS899)